File: app/src/main/java/com/theandroiddev/mywins/MyWinsApplication.java
package com.theandroiddev.mywins;

import android.app.Application;

import com.theandroiddev.mywins.injection.AppComponent;
import com.theandroiddev.mywins.injection.AppModule;
import com.theandroiddev.mywins.injection.DaggerAppComponent;

/**
 * Created by jakub on 07.11.17.
 */ 

public class MyWinsApplication extends Application {

    private AppComponent appComponent;

    @Override

    public void onCreate() {
        super.onCreate();
        appComponent = initDagger(this);
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }

    protected AppComponent initDagger(MyWinsApplication application) {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(application))
                .build();
    }

}

File: app/src/main/java/com/theandroiddev/mywins/MyWinsApplication.kt
package com.theandroiddev.mywins

import com.theandroiddev.mywins.injection.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication

/**
 * Created by jakub on 07.11.17.
 */

class MyWinsApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val component = DaggerAppComponent.builder().application(this).build()
        component.inject(this)
        return component
    }

}
